src/java/com/sarality/app/loader/AsyncResponse.java

```diff
@@ -0,0 +1,10 @@
+package com.sarality.app.loader;
+
+/**
+ * Async Response interface to notify the calling entity when the asyncTask is completed
+ *
+ * @author Sunayna Uberoy
+ */
+public interface AsyncResponse<T> {
```

Comment:
I am not using the requestCode in the implementation. i will remove it. it was only for future implementation.


On 24-Feb-2015, at 1:27 pm, Abhideep Singh <notifications@github.com> wrote:

> In src/java/com/sarality/app/loader/AsyncResponse.java:
> 
> > @@ -0,0 +1,10 @@
> > +package com.sarality.app.loader;
> > +
> > +/**
> > + * Async Response interface to notify the calling entity when the asyncTask is completed
> > + *
> > + * @author Sunayna Uberoy
> > + */
> > +public interface AsyncResponse<T> {
> Is this a Response or a TaskCompletionListener? A better name for this might be something like OnTaskCompleteListener
> 
> I also don't get why requestCode needs to be passed in as a method parameter. You either follow the ActivityResult model or do Activity chaining.
> 
> In activity chaining
> 1) You call Show E-ticket Fragment / Activity which simply tries to load the data from file.
> 2) If the eticket does not exist as a file on the sd-card, this activity calls the DownloadETicketActivity or Fragment. This runs a task in the background to load the file from Gmail and save it on file. As part of this activity we display the Gmail Authorization and AccountPicker which can follow the ActivityReponse model.
> 
> In these scenarios I don't see why we need the requestcode. Success or failure perhaps, but not much else.
> 
> Will address that more in the actual code review which does the file downloading
> 
> —
> Reply to this email directly or view it on GitHub.
>